README.md

```diff
@@ -1,2 +1,123 @@
 # autobus
+
 Bus for Node &amp; Browserify
+
+-   [Purpose](#purpose)
+-   [Installation](#installation)
+-   [Usage](#usage-example)
+-   [Methods](#methods)
+    -   [join](#joinchannel-listener)
+    -   [send](#sendchannel-message)
+    -   [publish](#publishchannel-message)
+    -   [leave](#leavechannel-listener)
+
+## Purpose
+
+Send messages between channels and add listener functions to receive messages.
+
+## Installation
+
+```shell
+npm install autobus
+```
+
+## Usage Example
+
+```typescript
+import AutoBus from 'autobus';
+
+interface ChatMessage {
+    text: string;
+    user: string;
+    date: Date;
+}
+
+const displayMessage = (message: ChatMessage) => {
+    const { text, user, date } = message;
+    const formattedDate = date.toTimeString().split(' ')[0];
+    console.log(`[${formattedDate}] ${user}: ${text}`);
+};
+
+class ChatSession {
+    constructor(autoBus: AutoBus, user: string) {
+        this.autoBus = autoBus;
+        this.user = user;
+        this.init();
+    }
+
+    init() {
+        this.autoBus.join('chat.channel', this.onMessageReceived.bind(this));
+    }
+
+    onMessageReceived(message: ChatMessage) {
+        if (message.user === this.user) {
+            return;
+        }
+        displayMessage(message);
+    }
+
+    publishMessage(message: string) {
+        const chatMessage: ChatMessage = {
+            text: message,
+            user: this.user,
+            date: new Date()
+        };
+        this.autoBus.publish('chat.channel', chatMessage);
+    }
+}
+
+const autoBus = new AutoBus();
+
+const session1 = new ChatSession(autoBus, 'User1');
+const session2 = new ChatSession(autoBus, 'User2');
+
+session1.publishMessage("Hello I'm User1");
+session2.publishMessage("Hello I'm User2");
+```
+
+#### Example output
+
+```
+[12:30:28] User1: Hello I'm User1
+[12:30:28] User2: Hello I'm User2
+```
+
+## Methods
+
+### join(channel, listener)
```

Comment:
Great. Can you also add a subheader to `join` with one example with wild card matching? Such as:

```ts
autobus.join('*.channel', consoleLogData);
```